connectors/byoc.py

```diff
@@ -27,6 +27,8 @@
 PIPELINE = "ent-search-generic-ingestion"
 SYNC_DISABLED = -1
 DEFAULT_ANALYSIS_ICU = False
+ON_DEMAND = 'on-demand'
```

Comment:
if these values are part of the protocol, lets use an Enum, like Status and JobStatus:

```
class TriggerType(Enum):
  ON_DEMAND = 1
  SCHEDULED = 2
```

Can we also change `on-demand` to `on_demand` in the protocol to stay consistent with the used notation (`in_progress`, `starts_with` etc) cc @sphilipse